### PR TITLE
airbyte-ci: only install main dependencies when calling poetry install

### PR DESCRIPTION
**Pull request by alafanechere**
## What
We used the classic `poetry install` command in a container to install a package.
This commands also installs dev dependencies, which is not something we want when no additional depedencie groups are passed to the function
## How
Run `poetry install --only main` when no `additional_dependency_groups` is passed to the `_install_python_dependencies_from_poetry` function.

### CODE REVIEW

**Comment by alafanechere** [comment]
* `master`
  * **#34945** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/34945?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py [hunk 1]
```diff
@@ -159,12 +159,13 @@ def _install_python_dependencies_from_poetry(
 ) -> Container:
     pip_install_poetry_cmd = ["pip", "install", "poetry"]
     poetry_disable_virtual_env_cmd = ["poetry", "config", "virtualenvs.create", "false"]
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py:161]: Does this imply that inside of the connector images, poetry will install deps into global python package space?

Comment (natikgadzhi) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py:161]: I think this allows us to then run the connector with the default entry point, as opposed to having to do `poetry run whatever.py`, right?

**Review by natikgadzhi — APPROVED** [review]

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py (hunk 1 above)
Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py:161]: > Does this imply that inside of the connector images, poetry will install deps into global python package space?

Yes, building the docker image is a two step approach:
* We provision a builder container on which we run the package installation: this is where poetry is a requirement
* We mount the installed packages to a new container which is transformed into our final image

This means that `poetry` could eventually not be bundled in our final images, the build dependencies being required only in the builder container.

But for simplicity the builder and the final connector image use the same base image: our python connector base image which has poetry.

If we'd want to optimize the image size we could just install build dependencies in the builder container.

**Pull request merged**